src/realm/util/serializer.cpp

```diff
@@ -99,6 +100,95 @@ std::string print_value<>(realm::Timestamp t)
     return ss.str();
 }
 
+
+// The variable name must be unique with respect to the already chosen variables at
+// this level of subquery nesting and with respect to the names of the columns in the table.
+// This assumes that columns can start with '$' and that we might one day want to support
+// referencing the parent table columns in the subquery. This is currently disabled by an assertion in the
+// core SubQuery constructor.
+std::string SerialisationState::get_variable_name(ConstTableRef table) {
+    std::string guess_prefix = "$";
+    const char start_char = 'x';
+    char add_char = start_char;
+
+    auto next_guess = [&]() {
+        add_char = (((add_char + 1) - 'a') % 26) + 'a';
+        if (add_char == start_char) {
+            guess_prefix += add_char;
+        }
+    };
+
+    while (true) {
```

Comment:
Missing assert or other handling if the column is never found (should be there even if it by design can't happen)

Reply:
By design it will loop endlessly until a unique name is found. I assume there must be a finite number of column names to search through and a finite number of subquery nestings with different variable names. In practice though it'll find a match on the first pass. Would you suggest putting an upper limit on the number of loop iterations we allow?

Reply:
oh okay, I misread the code and thought it would loop until it had somehow found a specific column :) no worries then